File: src/game/tools/rpc/output/wrap/TopListWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type TopListWrap struct {
	
	Kind int16 // 类型
	List []*TopRankWrap // 数据
	rpc.Wrapper
}

func (w *TopListWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.Kind = pck.PopInt16(); 
	w.List = make([]*TopRankWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.List); i++ {
		w.List[i] = new(TopRankWrap)
		w.List[i].Decode(pck)
		
	}
	
	return w
}

func (w *TopListWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt16(w.Kind) 
	if w.List == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.List)))
    	for i := 0; i < len(w.List); i++ {
    		w.List[i].Encode(pck);
			
    	}
    }
	
}

File: src/game/tools/rpc/output/wrap/TopListWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class TopListWrap : IWrapper
	{	
		public short Kind; // 类型
		public TopRankWrap[] List; // 数据
		
		public void Decode(Packet pck)
		{	
			Kind = pck.GetShort(); 
			List = new TopRankWrap[pck.GetShort()];
			for (int i = 0; i < List.Length; i++)
			{
				List[i] = new TopRankWrap();
				List[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutShort(Kind); 
        	if (List == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)List.Length);
	        	for(int i = 0; i < List.Length; i++)
	        	{
	        		List[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}